core/auth.go

```diff
@@ -56,25 +56,34 @@ type authResponse struct {
 	Tenants []authTenant `json:"tenants"`
 }
 
-func (core *Core) checkAuth(req *http.Request) (*authResponse, error) {
-	cookie, err := req.Cookie(SessionCookieName)
-	if err != nil {
-		if err == http.ErrNoCookie {
-			log.Debugf("no session cookie ('%s') found in request; treating as unauthenticated", SessionCookieName)
-		} else {
-			log.Debugf("failed to extract session cookie ('%s') from request: %s", SessionCookieName, err)
+//Gets the session ID from the request. Returns "" if not given.
+func getSessionID(req *http.Request) string {
+	sessionID := req.Header.Get("X-Shield-Session")
+
+	if sessionID == "" { //If not in header, check for cookie
+		cookie, err := req.Cookie(SessionCookieName)
+		if err == nil { //ErrNoCookie is the only error returned from Cookie()
```

Comment:
I would believe that now, but in the future?

Reply:
"Cookie returns the named cookie provided in the request or ErrNoCookie if not found. "